File: Leetcode/30_Days_Pandas/3.py
# https://leetcode.com/problems/customers-who-never-order/?envType=study-plan-v2&envId=30-days-of-pandas&lang=pythondata
# COMPLETE

import pandas as pd

def find_customers(customers: pd.DataFrame, orders: pd.DataFrame) -> pd.DataFrame:
        return customers[~customers['id'].isin(orders['customerId'])][['name']].rename(columns={'name': 'Customers'})

File: Leetcode/30-Days-JS/3.js
// https://leetcode.com/problems/to-be-or-not-to-be/?envType=study-plan-v2&envId=30-days-of-javascript
// COMPLETE

/**
 * @param {any} val
 * @return {Object}
 */
var expect = function(val) {
    return {
        toBe: function(expected) {
            if (val === expected) {
                return true;  // Directly return true
            } else {
                throw new Error("Not Equal");
            }
        },
        notToBe: function(expected) {
            if (val !== expected) {
                return true;  // Directly return true
            } else {
                throw new Error("Equal");
            }
        }
    };
};
